packages/pages-components/src/components/analytics/Analytics.ts

```diff
@@ -30,54 +26,35 @@ import { getPartition } from "../../util/partition.js";
  */
 export class Analytics implements AnalyticsMethods {
   private _optedIn: boolean;
-  private _conversionTrackingEnabled = false;
-  private _cookieManager: CookieManager | undefined;
-  private _analyticsReporter: PagesAnalyticsService | undefined;
+  private _sessionTrackingEnabled: boolean;
+  private _analyticsEventService: AnalyticsEventService | undefined;
   private _pageViewFired = false;
-  private _enableDebugging = false;
 
   /**
    * Creates an Analytics instance, will fire a pageview event if requireOptin
    * is false
    *
+   * @param apiKey - the Yext API key for authorizing analytics events
+   * @param defaultCurrency - the ISO 4217 currency code to use for valued events
    * @param templateData - template data object from the pages system
    * @param requireOptIn - boolean, set to true if you require user opt in before tracking analytics
+   * @param disableSessionTracking - turns off session tracking
+   * @param enableDebugging - turns debug mode on meaning requests are logged instead
    */
   constructor(
+    private apiKey: string,
+    private defaultCurrency: string,
     private templateData: TemplateProps,
     requireOptIn?: boolean | undefined,
-    private pageDomain?: string,
-    private productionDomains: string[] = []
+    disableSessionTracking?: boolean | undefined,
```

Comment:
Optional params should be after the non-optional params.

Could we also set defaults for these like we do for enableDebugging?

Reply:
In this case it's nice to have enableDebugging last since it's something you'd turn on/off, the rest are configuration. Since this is a constructor it doesn't matter for order and also not used by anything other than the AnalyticsProvider.

I can't set a default for the non-private params.